python/pyxir/contrib/target/components/common/op_support.py

```diff
@@ -297,3 +297,21 @@ def is_scale_supported(
     axis = X.attrs["axis"]
     channels = X.shapes[axis]
     return channels >= 1 and channels <= 256 * channel_parallel
+
+
+def is_upscale_supported(
+    X: XLayer,
+    bXs: List[XLayer],
+    tXs: List[XLayer],
+    channel_parallel: int,
+    bank_depth: int,
+    bank_num: int,
+) -> bool:
+    method = X.attrs["method"] 
+    input_channel = X.shapes[X.attrs["data_layout"].index("C")]
+    scale_h = X.attrs["scale_h"]
+    scale_w = X.attrs["scale_w"]
+    return (math.ceil(scale_h / bank_num) * scale_h * math.ceil(input_channel / channel_parallel) <= bank_depth) \
```

Comment:
We can also support bilinear mode from the doc. Could you try adding that and corresponding tests?

Reply:
DPU supports only nearest_neighbor

[UNILOG][INFO] Compile mode: dpu
[UNILOG][INFO] Debug mode: function
[UNILOG][INFO] Target architecture: DPUCAHX8H_ISA2
[UNILOG][INFO] Graph name: quantize_eval_model, with op num: 11
[UNILOG][INFO] Begin to compile...
**[UNILOG][WARNING] xir::Op{name = upsample, type = upsample-fix} has been assigned to CPU: [DPU does not support BILINEAR mode. (only support NEAREST mode)]**